### PR TITLE
Fixing issue in retrieving payload from message context

### PR DESCRIPTION
**Pull request by Ashi1993**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java [hunk 1]
```diff
@@ -44,11 +46,29 @@ public class MT940FormatValidator extends AbstractConnector {
...
+                payload = jsonPayload.getJSONObject("text").getString("content");
+            } else {
+                for (Iterator it = jsonPayload.keys(); it.hasNext(); ) {
+                    String key = (String) it.next();
+                    if (key.startsWith("axis2ns")) {
+                        payload = jsonPayload.getJSONObject(key).getString("content");
+                    }
+                }
+            }
+        } catch (Exception e) {
```
Comment (Akila94) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:64]: Shall we catch the specific exception here?

**Review by aka4rKO — APPROVED** [review]

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java (hunk 1 above)
Comment (Ashi1993) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java]: Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/5/commits/ba0641e003d242353b2dba4d6583a7b390ac9f46

**Review by Akila94 — APPROVED** [review]

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java [hunk 2]
```diff
@@ -44,11 +47,29 @@ public class MT940FormatValidator extends AbstractConnector {
...
+            } else {
+                for (Iterator it = jsonPayload.keys(); it.hasNext(); ) {
+                    String key = (String) it.next();
+                    if (key.startsWith("axis2ns")) {
+                        payload = jsonPayload.getJSONObject(key).getString("content");
+                    }
+                }
+            }
+        } catch (JSONException e) {
+            log.error(String.format("Failed to read the text payload from request. %s", e.getMessage()));
```
Comment (VimukthiRajapaksha) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:66]: Don't we need to log the stackTrace here?

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java [hunk 3]
```diff
@@ -44,11 +47,29 @@ public class MT940FormatValidator extends AbstractConnector {
...
+            JSONObject jsonPayload = XML.toJSONObject(messageContext.getEnvelope().getBody().toString())
+                    .getJSONObject("soapenv:Body");
+            if (jsonPayload.has("text")) {
+                payload = jsonPayload.getJSONObject("text").getString("content");
+            } else {
+                for (Iterator it = jsonPayload.keys(); it.hasNext(); ) {
+                    String key = (String) it.next();
+                    if (key.startsWith("axis2ns")) {
+                        payload = jsonPayload.getJSONObject(key).getString("content");
+                    }
```
Comment (VimukthiRajapaksha) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:62]: Can we break the loop here? also its clean if we could use streams for this
```suggestion
                    if (key.startsWith("axis2ns")) {
                        payload = jsonPayload.getJSONObject(key).getString("content");
                        break;
                    }
```

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java [hunk 4]
```diff
@@ -44,11 +47,29 @@ public class MT940FormatValidator extends AbstractConnector {
...
-                .getAxis2MessageContext();
-        Optional<String> payload = ConnectorUtils.buildMessagePayloadFromMessageContext(axis2MessageContext);
-        if (payload.isPresent()) {
-            String[] lines = payload.get().split(ConnectorConstants.LINE_BREAK);
+
+        String payload = null;
+        try {
+            JSONObject jsonPayload = XML.toJSONObject(messageContext.getEnvelope().getBody().toString())
+                    .getJSONObject("soapenv:Body");
+            if (jsonPayload.has("text")) {
```
Comment (VimukthiRajapaksha) [inline src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:55]: could `jsonPayload` be null?

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java (hunk 3 above)
Comment (Ashi1993) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:62]: Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/7

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java (hunk 4 above)
Comment (Ashi1993) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:55]: Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/7

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java (hunk 2 above)
Comment (Ashi1993) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:66]: Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/7

File: src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java [hunk 5]
```diff
@@ -44,11 +47,29 @@ public class MT940FormatValidator extends AbstractConnector {
...
+                payload = jsonPayload.getJSONObject("text").getString("content");
```
[the remark this answers is not in the document]
Comment (Ashi1993) [reply src/main/java/org/wso2/carbon/module/swiftiso20022/MT940FormatValidator.java:56]: Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/7

**Pull request merged**